torchgeo/datasets/ssl4eo.py

```diff
@@ -51,7 +225,10 @@ class SSL4EOS12(NonGeoDataset):
 
     size = 264
 
-    metadata = {
+    _Metadata = TypedDict(
+        "_Metadata", {"filename": str, "md5": str, "bands": list[str]}
+    )
```

Comment:
TypedDict allows us to remove all uses of Any and cast! We should try to use this in more places.